lib/helper/user_comms.rb

```diff
@@ -0,0 +1,25 @@
+module Emerald
+  class UserCommsHelper
+
+    ERROR_INCORRECT_FILE_TYPE = "Oops! Incorrect file type received.\nExpecting a lisp file e.g. example.lisp\n"
+    ERROR_INITIALISE_WITH_STRING_IO = "Error: Initialize with StringIO objects"
+
+    def initialize(stdout)
+      raise ERROR_INITIALISE_WITH_STRING_IO if stdout.nil?
+      @stdout = stdout if stdout.respond_to?(:puts)
```

Comment:
It's a way of validating that the correct argument was passed in, but I also didn't want this class to be too rigidly bound to interacting with the user only through the terminal. And if in future we wanted it to handle user input as well, I could have a check that such as:
```ruby
stdin.respond_to?(:gets)
```
to verify that the arguments had been passed in the correct order. Also I think this check should actually be before the nil check, so that it's not assigned unless it responds to puts, etc